### PR TITLE
Check requestState before running post scan scripts

### LINKED ISSUE #74: Check requestState before running post scan scripts
Before running this step , the script should check that the scan `requestState` is updated / finished so that the ws_sbom_generator won't run on an empty project.
I want to recommend on this yml in this [issue](https://github.com/whitesource-ps/ws-sbom-generator/issues/80) , but such validation is missing.
 
There is a [validation](https://github.com/whitesource-ps/ws-sbom-generator/blob/ccb5ae6b0eb2ff7d0b166a7f02f79af84a5bedcc/ws_sbom_generator/sbom_generator.py#L72) for empty project , but its relevant for case where no libraries were resolved from the scan.

image

_Originally posted by @rammatzkvosky in https://github.com/whitesource-ft/ws-examples/pull/44#discussion_r813774690_

### PR DESCRIPTION
**Pull request by tabac-ws**
Fixes #74
@samq-ws -pls review

### CODE REVIEW

File: Scripts/whitesource-spdx.yml [hunk 1]
```diff
@@ -45,6 +45,32 @@ jobs:
             echo WS_URL: $WS_URL
             echo === Installing WhiteSource SBOM Generator ===
             pip  install ws_sbom_generator
+            echo === Checking Project State before generating SBOM ===
+            IFS="|"
+            scan_status=true
+            pass_status=("UPDATE"${IFS}"FINISH"${IFS}"DIFF")
+            fail_status=("UNKONWN"${IFS}"FAIL")
+            scan_status=true
```
Comment (samq-ws) [inline Scripts/whitesource-spdx.yml:53]: section 50-55 is duplicated...remove one set

File: Scripts/whitesource-spdx.yml [hunk 2]
```diff
@@ -45,6 +45,32 @@ jobs:
...
+            pass_status=("UPDATE"${IFS}"FINISH"${IFS}"DIFF")
+            fail_status=('UNKONWN'${IFS}'FAIL')
+            while $scan_status
+            do
+              new_status=$(curl -s -X POST -H "Content-Type: application/json" -d '{"requestType":"getProjectState", "userKey": "$WS_USERKEY", "projectToken":"$WS_PROJECTTOKEN"}' $WS_URL/api/v1.3 | jq '.projectState|.lastProcess' | tr -d '"')
+              if [[ "${IFS}${pass_status[*]}${IFS}" =~ "${IFS}${new_status}${IFS}" ]];
+              then
+                      scan_status=false
+                      echo "Project infromation has been uploaded successfully!"
```
Comment (samq-ws) [inline Scripts/whitesource-spdx.yml:62]: information is spelled incorrectly

File: Scripts/whitesource-spdx.yml [hunk 3]
```diff
@@ -45,6 +45,32 @@ jobs:
             echo WS_URL: $WS_URL
             echo === Installing WhiteSource SBOM Generator ===
             pip  install ws_sbom_generator
+            echo === Checking Project State before generating SBOM ===
+            IFS="|"
+            scan_status=true
+            pass_status=("UPDATE"${IFS}"FINISH"${IFS}"DIFF")
+            fail_status=("UNKONWN"${IFS}"FAIL")
```
Comment (samq-ws) [inline Scripts/whitesource-spdx.yml:52]: UNKNOWN is spelled incorrectly....is this a bug in WSS or a typo?

File: Scripts/whitesource-spdx.yml [hunk 4]
```diff
@@ -45,6 +45,32 @@ jobs:
...
+            scan_status=true
+            pass_status=("UPDATE"${IFS}"FINISH"${IFS}"DIFF")
+            fail_status=('UNKONWN'${IFS}'FAIL')
+            while $scan_status
+            do
+              new_status=$(curl -s -X POST -H "Content-Type: application/json" -d '{"requestType":"getProjectState", "userKey": "$WS_USERKEY", "projectToken":"$WS_PROJECTTOKEN"}' $WS_URL/api/v1.3 | jq '.projectState|.lastProcess' | tr -d '"')
```
Comment (samq-ws) [inline Scripts/whitesource-spdx.yml:58]: need to surround variables with single quotes
```"'$WS_USERKEY'"```
```"'$WS_PROJECTTOKEN'"```

**Review by samq-ws — CHANGES_REQUESTED** [review]
this should a be separate script as this is valuable for all of our post-scan scripts such as prioritize, report generation, etc.

File: Scripts/whitesource-spdx.yml (hunk 3 above)
Comment (tabac-ws) [reply Scripts/whitesource-spdx.yml:52]: Typo

**Review by samq-ws — APPROVED** [review]

**Pull request merged**